internal/tool/generate/generator.go

```diff
@@ -422,6 +423,27 @@ func extractComponent(opt Options, pkg *packages.Package, file *ast.File, tset *
 					formatType(pkg, arg))
 			}
 			refs = append(refs, named)
+		} else if isWeaverListener(t) {
+			var lisName string
+			if f.Tag != nil {
+				tag := f.Tag.Value
```

Comment:
Would [`reflect.StructTag(tag).Get("weaver")`](https://pkg.go.dev/reflect#StructTag.Get) work here?

Reply:
Unfortunately, it wouldn't, since this `Tag` includes the single quotes. However, I just realized that the user may attach other tags (e.g., ```x weaver.Listener `weaver:"foo",json:"bar"` ```). I'll change this in a separate PR.